File: src/1413-Minimum-Value-to-Get-Positive-Step-by-Step-Sum/minimum_value_to_get_positive_step_by_step_sum.py
from typing import List

class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        min_start: int = 0
        curr_sum: int = 0
        for n in nums:
            curr_sum += n
            min_start = max(min_start, 1 - curr_sum)

        return min_start if min_start > 0 else 1

if __name__ == "__main__":
    nums: List[int] = [-3,2,-3,4,2]
    
    solution = Solution()

    print(solution.minStartValue(nums))
    

File: src/1413-Minimum-Value-to-Get-Positive-Step-by-Step-Sum/minimum_value_to_get_positive_step_by_step_sum.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution {
public:
    int minStartValue(std::vector<int>& nums) {
        int min_start = 1;
        int curr_sum = 0;

        for(auto n : nums) {
            curr_sum += n;
            min_start = std::max(min_start, 1 - curr_sum);
        }

        if (min_start > 0) return min_start;
        else return 1;
    }
};

int main() {
    std::vector<int> nums = {-3,2,-3,4,2};

    Solution solution;
    std::cout << solution.minStartValue(nums);

    return 0;
}